src/logic/basic.lean

```diff
@@ -1136,6 +1136,9 @@ by simp only [← @forall_eq _ p a, ← forall_and_distrib, ← or_imp_distrib,
 @[simp] theorem forall_eq_or_imp {a' : α} : (∀ a, a = a' ∨ q a → p a) ↔ p a' ∧ ∀ a, q a → p a :=
 by simp only [or_imp_distrib, forall_and_distrib, forall_eq]
 
+lemma ne.ne_of_ne {x y z : α} (h : x ≠ y) : x ≠ z ∨ y ≠ z :=
```

Comment:
I still think dot notation isn't appropriate, particularly since the `simp` lemma `ne.def` often ruins it.

Reply:
`ne.symm` would disagree with that claim, although perhaps we should have an alias under the non-dotted name. Do you have any examples in mind of lemmas about `ne` that could use dot notation but don't, @vihdzp?

Reply:
The name I suggested was `ne.ne_or_ne`; I don't think `ne.ne_of_ne` makes sense ...

Reply:
I updated the name.

I have also experienced that `ne.symm` does not always work with dot-notation, but do not have an example at hand.  I am wondering if it is an issue between `a ≠ b` and `¬a = b`?

Reply:
Well, I guess dot notation is fine then.